python_modules/dagster-webserver/dagster_webserver_tests/webserver/test_app.py

```diff
@@ -278,3 +280,85 @@ def test_download_compute(instance, test_client: TestClient):
 
     response = test_client.get(f"/download/{run_id}/jonx/stdout")
     assert response.status_code == 404
+
+
+def test_runless_events(instance: DagsterInstance, test_client: TestClient):
+    # base case
+    my_asset_key = "my_asset"
+    response = test_client.post(f"/report_asset_materialization/{my_asset_key}")
+    assert response.status_code == 200
+    evt = instance.get_latest_materialization_event(AssetKey(my_asset_key))
+    assert evt
+
+    # empty asset key path
+    response = test_client.post("/report_asset_materialization/")
+    assert response.status_code == 400
+
+    # multipart key
+    long_key = AssetKey(["foo", "bar", "baz"])
+    response = test_client.post(
+        f"/report_asset_materialization/{long_key.to_user_string()}",
+    )
+    assert response.status_code == 200
+    evt = instance.get_latest_materialization_event(long_key)
+    assert evt
```

Comment:
multipart key test here

Reply:
1) can you add assertion against fully expanded url (e.g. assert url == "report_asset_materialization/foo/bar/baz")
2) can you add test when asset key components contain "/"

I'm concerned about the answers to 2.

Reply:
opted to support accepting `asset_key` via query params or post body to provide ways to workaround `/` in the asset key content. Could not find a clean path trying to deal url encoded `/` values being part of the path since the web framework resolves those encodings